src/rtp/srtp.rs

```diff
@@ -109,37 +196,65 @@ impl SrtpContext {
         header: &RtpHeader,
         srtp_index: u64, // same as ext_seq
     ) -> Option<Vec<u8>> {
-        if buf.len() < SRTP_HMAC_LEN {
-            return None;
-        }
+        match &mut self.rtp {
+            Derived::Aes128CmSha1_80 {
+                hmac, salt, dec, ..
+            } => {
+                use aes_128_cm_sha1_80::{RtpHmac, ToRtpIv, HMAC_TAG_LEN};
+
+                if buf.len() < HMAC_TAG_LEN {
+                    return None;
+                }
 
-        let hmac_start = buf.len() - SRTP_HMAC_LEN;
+                let hmac_start = buf.len() - HMAC_TAG_LEN;
 
-        if !self
-            .rtp
-            .hmac
-            .rtp_verify(&buf[..hmac_start], srtp_index, &buf[hmac_start..])
-        {
-            trace!("unprotect_rtp hmac verify fail");
-            return None;
-        }
+                if !hmac.rtp_verify(&buf[..hmac_start], srtp_index, &buf[hmac_start..]) {
+                    trace!("unprotect_rtp hmac verify fail");
+                    return None;
+                }
+
+                let iv = salt.rtp_iv(*header.ssrc, srtp_index);
+
+                let input = &buf[header.header_len..hmac_start];
+                // Allocate enough to also hold a header, since this is used in rtp-mode.
+                let mut output = Vec::with_capacity(buf.len());
```

Comment:
This can be replaced with `vec![0; buf.len()]`

Reply:
yes